platform/initialize.go

```diff
@@ -0,0 +1,144 @@
+package platform
+
+import (
+	"context"
+	"github.com/kiegroup/container-builder/api"
+	"github.com/kiegroup/container-builder/client"
+	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/builder"
+	"github.com/kiegroup/kogito-serverless-operator/constants"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewInitializeAction returns a action that initializes the platform configuration when not provided by the user.
+func NewInitializeAction() Action {
+	return &initializeAction{}
+}
+
+type initializeAction struct {
+	baseAction
+}
+
+func (action *initializeAction) Name() string {
+	return "initialize"
+}
+
+func (action *initializeAction) CanHandle(platform *v08.KogitoServerlessPlatform) bool {
+	return platform.Status.Phase == "" || platform.Status.Phase == v08.PlatformPhaseDuplicate
```

Comment:
Phase == None

Why platform phase duplicate is a valid condition to proceed?

Reply:
If we skipped the platform initialization because there is another platform with the same traits and we delete the active one, we can resume the initialization of the platform that we previously skipped due to the duplication.